Eastern-Research-Group/csb-rebate-forms-app
# Feature/update app to support multi years

## Related Issues:
* CSBAPP-194

## Main Changes:
Major update to the app to allow for the new 2023 rebate year of submissions. Initially supports the new 2023 FRF form, but lays the groundwork for the remaining two 2023 forms, and future rebate year forms.

## Steps To Test:
1. Navigate to the dashboard.
2. Change "Rebate Year" to 2023, and ensure the form submissions shown below reflect the new rebate year (will display the "Please select the “New Application” button..." message initially until you create your first 2023 FRF submission).
3. Create a new FRF submission for 2023, and ensure you're re-directed to the correct page to continue editing the form.
4. Navigate back to the dashboard, and switch "Rebate Year" back to 2022, and ensure 2022 form submissions display properly.
5. Navigate directly to an old 2022 form submission via the old URL (e.g. "/rebate/644034b7ed3e46078d54cc99") and ensure you're redirected to the new URL (e.g. "/frf/2022/644034b7ed3e46078d54cc99")

## TODO:
- [ ] Nothing to do until post-v4 release when the BAP has built out their 2023 FRF functionality, but eventually we'll want to update the app to fetch 2023 FRF data from the BAP and update the setting of the 2023 FRF submission statuses to account for that.
